### PR TITLE
fix(clerk-js): Update cache key for roles when organization changes

### PR DESCRIPTION
**Pull request by panteliselef**
## Description
Fixes SDK-1822

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

**Review by LauraBeatris — APPROVED** [review]

File: packages/clerk-js/src/ui/components/OrganizationSwitcher/__tests__/OrganizationSwitcher.test.tsx [hunk 1]
```diff
@@ -128,6 +128,7 @@ describe('OrganizationSwitcher', () => {
         f.withOrganizations();
         f.withUser({ email_addresses: ['test@clerk.com'], organization_memberships: ['Org1', 'Org2'] });
       });
+      fixtures.clerk.organization?.getRoles.mockRejectedValue(null);
```
Comment (dimkl) [inline packages/clerk-js/src/ui/components/OrganizationSwitcher/__tests__/OrganizationSwitcher.test.tsx:131]: ❓ @panteliselef Based on the ticket i think that we need to add a test to verify that different organization roles are displayed / fetched when the organization is switched.
The updated test does not seem to be related to that scenario.

File: packages/clerk-js/src/ui/hooks/useFetchRoles.ts [hunk 2]
```diff
@@ -13,7 +13,10 @@ const getRolesParams = {
 };
 export const useFetchRoles = (enabled = true) => {
   const { organization } = useOrganization();
-  const { data, isLoading } = useFetch(enabled ? organization?.getRoles : undefined, getRolesParams);
+  const { data, isLoading } = useFetch(enabled && !!organization?.id ? organization?.getRoles : undefined, {
+    ...getRolesParams,
+    orgId: organization?.id,
+  });
```
Comment (dimkl) [inline packages/clerk-js/src/ui/hooks/useFetchRoles.ts:19]: 🙃 Do we populate the `organization` instance for personal account? If not i would expect to use `enabled && !!organization ` instead. Eg:
```typescript
  const { data, isLoading } = useFetch(enabled && !!organization ? organization?.getRoles : undefined, {
    ...getRolesParams,
    orgId: organization?.id,
  });
```
